core/rawdb/accessors_state.go

```diff
@@ -93,3 +93,57 @@ func DeleteCode(db ethdb.KeyValueWriter, hash common.Hash) {
 		utils.Logger().Error().Err(err).Msg("Failed to delete contract code")
 	}
 }
+
+// ReadValidatorCode retrieves the validator code of the provided code hash.
+func ReadValidatorCode(db ethdb.KeyValueReader, hash common.Hash) []byte {
+	// Try with the prefixed code scheme first, if not then try with legacy
+	// scheme.
+	data := ReadValidatorCodeWithPrefix(db, hash)
+	if len(data) != 0 {
+		return data
+	}
+	data, _ = db.Get(hash.Bytes())
+	return data
+}
+
+// ReadValidatorCodeWithPrefix retrieves the validator code of the provided code hash.
+// The main difference between this function and ReadValidatorCode is this function
+// will only check the existence with latest scheme(with prefix).
+func ReadValidatorCodeWithPrefix(db ethdb.KeyValueReader, hash common.Hash) []byte {
+	data, _ := db.Get(validatorCodeKey(hash))
+	return data
+}
+
+// HasValidatorCode checks if the validator code corresponding to the
+// provided code hash is present in the db.
+func HasValidatorCode(db ethdb.KeyValueReader, hash common.Hash) bool {
+	// Try with the prefixed code scheme first, if not then try with legacy
+	// scheme.
+	if ok := HasValidatorCodeWithPrefix(db, hash); ok {
```

Comment:
Why not simply use golang's in-built short circuiting? `return HasValidatorCodeWithPrefix(db, hash) || db.Has(Hash.Bytes())`